docs/building_images_with_bazel.md

```diff
@@ -0,0 +1,300 @@
+---
+tilt: Building Images with Bazel
+layout: docs
+---
+
+# Building Images with Bazel
+
+There's nothing more frustrating than using a tool that works perfectly except for _one_ small thing. Even if the tool is open source it can be a daunting task to get a change accepted to it.
+
+We recognize that everyone's set up is different and that we cannot foresee all possible features and use cases that folks will need. Software is complicated! As a result we designed Tilt to be customizable from the beginning. If there's functionality you wish Tilt had you can often add it yourself, right in your Tiltfile, no external pull requests necessary!
+
+Let's explore this with a simple feature request: I want Tilt to support the [Bazel](https://bazel.build/) build system. I can hear you now: but that's not simple at all! It's actually a lot simpler than you'd think, let's see why.
+
+_Note: Before diving in to this guide you should have run through [Tilt's Getting Started Guide](tutorial.html) and the deep dive in to [Tiltfile concepts](tiltfile_concepts.html). A familiarity with the Bazel build system will also clear a lot of things up but isn't necessary. Ready? Let's get started!_
```

Comment:
the "A familiarity with the Bazel build system" feels a bit weird if this doc is now "how to use tilt with your existing bazel build" rather than "here's an example of implementing a thing in the Tiltfile!"